gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/service/JavaServiceDistributionPlugin.java

```diff
@@ -217,12 +218,16 @@ public void apply(Project project) {
             task.setGroup(JavaServiceDistributionPlugin.GROUP_NAME);
             task.setDescription("Runs the specified project using configured mainClass and with default args.");
             task.dependsOn("jar");
-            task.doFirst(new Action<Task>() {
-                @Override
-                public void execute(Task _task) {
-                    task.setMain(mainClassName.get());
-                }
-            });
+            if (GradleVersion.current().compareTo(GradleVersion.version("6.4")) < 0) {
+                task.doFirst(new Action<Task>() {
+                    @Override
+                    public void execute(Task _task) {
+                        task.setMain(mainClassName.get());
+                    }
+                });
+            } else {
+                task.getMainClass().set(mainClassName);
```

Comment:
would this just throw a NoSuchMethodError if we ran it on gradle 6.3?

Reply:
yup